src/plugins/discord/mirror.js

```diff
@@ -0,0 +1,384 @@
+// @flow
+
+import type Database from "better-sqlite3";
+import stringify from "json-stable-stringify";
+import * as Model from "./models";
+import dedent from "../../util/dedent";
+
+const VERSION = "DISCORD_MIRROR_v1";
+
+/**
+ * Persists a local copy of data from a Discord Guild.
+ * Implements create and read functionality.
+ *
+ * Each Mirror instance is tied to a particular Guild. Trying to use a mirror
+ * for multiple Discord Guilds is not permitted; use separate Mirrors.
+ */
+
+export class Mirror {
```

Comment:
Tbh, it's kind of hard to get behind that naming convention. The SQLite aspect
 is the mirror; it feels like the Discord mirror should be called Mirror, and the implementation plan to create that mirror should be called something else.

Reply:
That's a good point. Arguably this is the Mirror. Where the other part is what takes the fetcher and mirror to figure out how to update the mirror with the latest data.

What I would suggest then, is to call it `SqliteMirror` still. Because I think when we're building this "how to update our mirror" class/function, we'll want to use an interface we can mock for tests.

In which case it would be: `interface Mirror`, `class SqliteMirror implements Mirror`,
`class MockMirror implements Mirror`, etc.

Reply:
Updated